editor/scene_tree_dock.cpp

```diff
@@ -292,6 +298,13 @@ void SceneTreeDock::_replace_with_branch_scene(const String &p_file, Node *base)
 	undo_redo->add_do_property(scene_tree, "set_selected", instantiated_scene);
 	undo_redo->add_undo_property(scene_tree, "set_selected", base);
 
+	if (EditorSettings::get_singleton()->get("docks/scene_tree/instantiate_with_editable_children")) {
+		undo_redo->add_do_method(edited_scene, "set_editable_instance", instantiated_scene, true);
+		undo_redo->add_undo_method(edited_scene, "set_editable_instance", false);
+		undo_redo->add_do_method(instantiated_scene, "set_display_folded", base->is_displayed_folded());
+		undo_redo->add_undo_method(instantiated_scene, "set_display_folded", false);
```

Comment:
Is this necessary? It makes folding part of the undo history (e.g. when you do this action when folded, unfold and do undo->redo, you'll end up folded again, because the folding was stored). IMO should be removed.